core/src/main/java/org/elasticsearch/index/query/MatchQueryParser.java

```diff
@@ -45,11 +62,9 @@ public MatchQueryBuilder fromXContent(QueryParseContext parseContext) throws IOE
         XContentParser parser = parseContext.parser();
 
         MatchQuery.Type type = MatchQuery.Type.BOOLEAN;
-        if ("match_phrase".equals(parser.currentName()) || "matchPhrase".equals(parser.currentName()) ||
-                "text_phrase".equals(parser.currentName()) || "textPhrase".equals(parser.currentName())) {
+        if (parseContext.parseFieldMatcher().match(parser.currentName(), MATCH_PHRASE_FIELD)) {
             type = MatchQuery.Type.PHRASE;
-        } else if ("match_phrase_prefix".equals(parser.currentName()) || "matchPhrasePrefix".equals(parser.currentName()) ||
-                "text_phrase_prefix".equals(parser.currentName()) || "textPhrasePrefix".equals(parser.currentName())) {
+        } else if (parseContext.parseFieldMatcher().match(parser.currentName(), MATCH_PHRASE_FIELD)) {
```

Comment:
Yes, it has to be MATCH_PHRASE_PREFIX_FIELD instead.

Thinking about the tests: This particular option points to an interesting problem - some of the query options exclude one another. So if over time we want to extend those tests to include all/many more/insert your own preference which I think we should, I need to think about a way to allow for generating more than one json string - correct?